.circleci/config.yml

```diff
@@ -26,6 +26,56 @@ jobs:
           paths:
             - node_modules
           key: v1-dependencies-{{ checksum "package.json" }}
-        
-      # run tests!
+
+  test:
+    docker:
+      - image: circleci/node:9.5
+
+    working_directory: ~/repo
+
+    steps:
+      - checkout
+      - restore_cache:
+          key: v1-dependencies-{{ checksum "package.json" }}
       - run: yarn test
+
+  flow:
+    docker:
+      - image: circleci/node:9.5
+
+    working_directory: ~/repo
+
+    steps:
+      - checkout
+      - restore_cache:
+          key: v1-dependencies-{{ checksum "package.json" }}
+      - run: yarn flow
+
+  lint:
+    docker:
+      - image: circleci/node:9.5
+
+    working_directory: ~/repo
+
+    steps:
+      - checkout
+      - restore_cache:
+          key: v1-dependencies-{{ checksum "package.json" }}
+      - run: yarn lint
+
+workflows:
+  version: 2
+
+  lint-flow-and-then-test:
+    jobs:
+      - build
+      - lint:
+          requires:
+            - build
+      - flow:
+          requires:
+            - build
+      - test:
+          requires:
+            - lint
```

Comment:
Shouldn't we remove this line? Lint issues should not prevent the tests from running, so we see both failures at the same time. It is ok for flow issues to prevent tests from running as they indicate coding mistakes that can break the tests.

Reply:
The lint job will only fail if we have lint errors, which are typically syntax errors. The warnings are being ignored, so they won't stop the tests.